tools/pyboard.py

```diff
@@ -240,7 +259,11 @@ def get_time(self):
 setattr(Pyboard, "exec", Pyboard.exec_)
 
 def execfile(filename, device='/dev/ttyACM0', baudrate=115200, user='micro', password='python'):
-    pyb = Pyboard(device, baudrate, user, password)
+    try:
+        pyb = Pyboard(device, baudrate, user, password)
+    except PyboardError as er:
+        print(er)
+        return
```

Comment:
Since execfile should be executed from within a python script I'd say it should raise an exception on error so that it can be handled by the caller.